packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -527,6 +529,91 @@ void main() {
     await tester.showKeyboard(find.byType(TextField));
     await tester.pump();
   });
+
+  group('getSemanticsData', () {
+    testWidgets('throws when there are no semantics', (WidgetTester tester) async {
+      await tester.pumpWidget(
+        new MaterialApp(
+          home: const Scaffold(
+            body: const Text('hello'),
+          ),
+        ),
+      );
+
+      expect(() => tester.getSemanticsData(find.text('hello')),
+        throwsA(isInstanceOf<StateError>()));
+    });
+
+    testWidgets('throws when there are multiple results from the finder', (WidgetTester tester) async {
+      final SemanticsHandle semanticsHandle = tester.ensureSemantics();
+
+      await tester.pumpWidget(
+        new MaterialApp(
+          home: new Scaffold(
+            body: new Row(
+              children: const <Widget>[
+                const Text('hello'),
+                const Text('hello'),
+              ],
+            ),
+          ),
+        ),
+      );
+
+      expect(() => tester.getSemanticsData(find.text('hello')),
+          throwsA(isInstanceOf<StateError>()));
```

Comment:
What's the actual error? Is the problem clear from that error?

Reply:
Changed so that specific messages are thrown when elements  == 0 and elements > 1